src/app/domains/wallets/domains.wallets.gateway.ts

```diff
@@ -1,42 +1,51 @@
-import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs';
-import { IWalletApiResponse, IWalletPayload } from './domains.wallets.types';
-import { fakeRequest } from '../../common/http/common.http.fake-request';
-import { getRandomNumber } from 'src/app/common/utils/common.utils.random';
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs';
+import {IWalletApiResponse, IWalletPayload} from './domains.wallets.types';
+import {fakeRequest} from '../../common/http/common.http.fake-request';
+import {getRandomNumber} from 'src/app/common/utils/common.utils.random';
+import {MyWallet} from '../../pages/wallets/my-wallets/pages-wallets-my-wallet.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class DomainsWalletsGateway {
   public getWallets(): Observable<IWalletApiResponse[]> {
-   return fakeRequest([
-     {
-       name: 'Wallet 1',
-       creationDate: '2022-10-22T09:47:52.595721658Z',
-       id: 1,
-     },
-     {
-       name: 'Wallet 2',
-       creationDate: '2022-10-23T09:47:52.595721658Z',
-       id: 2,
-     },
-     {
-       name: 'Wallet 3',
-       creationDate: '2022-10-24T09:47:52.595721658Z',
-       id: 3,
-     },
-   ])
+    return fakeRequest([
+      {
+        name: 'Wallet 1',
+        creationDate: '2022-10-22T09:47:52.595721658Z',
+        id: 1,
+      },
+      {
+        name: 'Wallet 2',
+        creationDate: '2022-10-23T09:47:52.595721658Z',
+        id: 2,
+      },
+      {
+        name: 'Wallet 3',
+        creationDate: '2022-10-24T09:47:52.595721658Z',
+        id: 3,
+      },
+    ])
   }
 
-  public addNewWallet({ name }: IWalletPayload): Observable<IWalletApiResponse> {
+  public addNewWallet({name}: IWalletPayload): Observable<IWalletApiResponse> {
     return fakeRequest({
       name: name,
       creationDate: '2022-10-22T09:47:52.595721658Z',
       id: getRandomNumber(100, 10000),
     });
   }
 
-  public updateNewWallet(id: number, { name }: IWalletPayload): Observable<IWalletApiResponse> {
+  public updateNewWallet(id: number, {name}: IWalletPayload): Observable<IWalletApiResponse> {
+    return fakeRequest({
+      name: name,
+      creationDate: '2022-10-23T09:47:52.595721658Z',
+      id: id,
+    });
+  }
+
+  public deleteWallet(id: number, {name}: IWalletPayload): Observable<IWalletApiResponse> {
```

Comment:
The response from the API will be empty with status `200`. That means that this method won't return `IWalletApiResponse` but `void` :)